### PR TITLE
[FIX] l10n_es_edi_tbai: workaround pos checkout w/ inv. printing

### PR DESCRIPTION
**Pull request by pivi-odoo**
## Issue
In the spanish localisation (specifically with `l10n_es_edi_tbai` installed), when generating an invoice at POS checkout, we are met with a stacktrace about failed formatting of empty date.

## Steps to reproduce
- Install l10n_es_edi_tbai, l10n_es, l10n_es_edii_sii, l10n_es_reports, point_of_sale
- Switch ES company
- Setup POS, create a new session
- Make an order and at checkout select "invoice" (this normally prints the invoice of the order)
- We have the stacktrace.

## Cause
In the spanish localisation, when posting an invoice, it is sent to the Basque government for a signature and for it to be added to a "blockchain" for proof of uniqueness. The invoices normally are sent to the government when being posted. Sadly currently there is no `l10n_pos_es` or equivalent at the moment, so when generating invoices when checking out in POS, we are trying generate the necessary tbai_id and qr_code for the pdf, but since the `signature` and `registration_date` are missing, we are met with a stacktrace (failed formatting of the `registration_date`, because it is empty). Subsequent try to print the invoice from the backend also leads to the same stacktrace.

## Fix
Don't print the id and qr code when the invoice is not signed.

## Affected versions
16 up to master

## Reference
opw-3414480

---
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

File: addons/l10n_es_edi_tbai/models/ir_actions_report.py
Comment (nle-odoo) [inline addons/l10n_es_edi_tbai/models/ir_actions_report.py:12]: might be ok (I don't know functionally), but it seems like this will be applied to all invoice, even if I have only a belgian company and I'm not  just installed l10n_es_ebi_tbai without installing it.

seeing the template below, it should probably also check a country code or l10n_es_tbai_is_required

**Review by nle-odoo — APPROVED** [review]
looks good to me if it's what we want with comment taken into account

note: If the intention is to have the added POT file translated in transifex, I think it also needs to be added in `/.tx/config` transifex config file.

File: addons/l10n_es_edi_tbai/models/ir_actions_report.py
Comment (pivi-odoo) [reply addons/l10n_es_edi_tbai/models/ir_actions_report.py:12]: You are right, I've extended the condition with l10n_es_tbai_is_required, since it checks everything that is needed :)

As for the addition of the .pot file into .tx/config, you mind checking what I've done, I emulated from the other entries, but it's the first time I am adding a .pot from scratch, just to be sure I've done what is needed. (I am adding the pot file, because I wanted to add the new term to be translatable, but I've noticed there is just no .pot for the module, so I've exported the whole of it and added it.

Comment (nle-odoo) [reply addons/l10n_es_edi_tbai/models/ir_actions_report.py:12]: looks good to me for the .tx/config (I've never done it)

1 typo: should be l10n_es_edi_tbai.pot and not l10n_multilang.pot (and alphabetically it should be before multilang)

**Review by nle-odoo — APPROVED** [review]

File: .tx/config
Comment (oomsveta) [inline .tx/config:1]: You should probably not add this module to the `.tx/config` 🙇

Except for Belgian localisations, l10n modules are not usually put on Transifex. The main reason for this is that we don't want people to start translating the Norwegian localisation into Chinese, for example, and Transifex doesn't offer the granularity to prevent this at any level other than a whole project, nor do we want to create a project by localisation 😕

File: .tx/config
Comment (pivi-odoo) [reply .tx/config:743]: yup '-.-

File: .tx/config
Comment (pivi-odoo) [reply .tx/config:1]: So should I remove the .pot completely also? or just the entry from .tx/config?

Comment (oomsveta) [reply .tx/config:1]: l10n modules often have a `.pot` file nevertheless, it's fine to keep it. As for me, I use it when I need to create or update corresponding `.po` files (which has to be done manually, as they are not on Transifex).

So you can just delete the entry from `.tx/config` 🙂 👌

**Comment by pivi-odoo** [comment]
Hello @oomsveta,

I've removed the entry in .tx/config. Now it's good to go, let me know if there is something else :)

**Review by oomsveta — APPROVED** [review]
looks ok to me, at least as far as translations are concerned 👌
it might be nice to add a Spanish translation for things like error messages that aren't directly translated in the sources, but since it's a bug fix, better not block this PR any longer; we can deal with this kind of stuff later

**Comment by pivi-odoo** [comment]
Hi @jco-odoo,

Would you mind making a review of this small bugfix?

Thanks in advance.

**Comment by jco-odoo** [comment]
@oomsveta It does not make sense to add only the pot here if we do not do it in Transifex.  The regular procedure is that we add at least the .po.  (could do pot as well, but only pot does not make any sense)

@pivi-odoo @nle-odoo It means that every time we do an order, we will get that error.  It is better than a traceback, but would not it be better if we could refer to the portal url instead e.g.  
cc @chklop

**Comment by jco-odoo** [comment]
@pivi-odoo The idea would be to simply be able to generate the invoice even if all the information is not there (so simply not show the barcode e.g.)

**Comment by pivi-odoo** [comment]
Hi @jco-odoo,

I've implemented your suggestion, instead of raising an error, the invoice is printed without the qr code.

File: addons/l10n_es_edi_tbai/views/report_invoice.xml
Comment (jco-odoo) [inline addons/l10n_es_edi_tbai/views/report_invoice.xml:5]: _is_in_chain might be enought.  

What we could do as well is to add L170-171 from _get_l10n_es_tbai_qr to _get_l10n_es_tbai_id instead. 

That way it is also fixed for those that do not update the module and it is consistent. 

@pivi-odoo Let me know what you think.

Comment (pivi-odoo) [reply addons/l10n_es_edi_tbai/views/report_invoice.xml:5]: So I've tested your proposition with a python only solution, sadly, it prints a visible QR code. So a change in the check in the xml is still necessary, there is no way around it :S 

I've still added the condition to `_get_l10n_es_tbai_id` as an additional assertion, it's better to have an `''` instead of a stacktrace if someone calls the method without the signature. (For now in standard `_get_l10n_es_tbai_id` is called only in this template, and when computing the qr code, which never happens if the invoice is not in the chain, so for now the condition is not really *needed*, but a *nice to have*).

I've also updated the condition in the xml as you've suggested, it's true it was redundant/overlapping.

Cheers.

**Pull request closed without merging**
